config/initializers/append_points.rb

```diff
@@ -25,6 +25,7 @@
 
 Workarea::Plugin.append_javascripts(
   'storefront.modules',
+  'workarea/storefront/paypal/modules/paypal',
```

Comment:
wondering if this module should be last. Assuming modules are initialized in order and the paypal sdk has already loaded, the `paypal` module will init, try to call all the `handlers` (there will be none), and then move on. Then, the other modules will init and add themselves to the handlers list, which will never get called until/if `WORKAREA.initModules` is called again for something else. Right?

Reply:
That's a good point, but also the other modules need to be able to call the `ready` method to add their handlers. Maybe running the handlers should be in a `setTimeout(..., 0)` or something?

Reply:
I should probably just modify `ready()` to call the function immediately if `paypalLoaded`.

Reply:
call ready isn't a problem if they are loaded before the `WORKAREA.paypal` because they load, then initialize. So `WORKAREA.paypal.ready` will exist before its initialization method is called

Reply:
I talked with Duffy on Slack about this a bit and we came to the conclusion that yes, this will most likely be a problem in the real world. To resolve this issue, the `init()` functions in the PayPal modules have been changed to just be calls to `WORKAREA.paypal.ready()`, which will call them on its own `init()` when PayPal has been loaded. This should also happen every time `initModules` is called, such as during an ajax request.

Reply:
Sounds good. Although it seems like to follow conventions, registering the PayPal stuff with `WORKAREA.paypal.ready` would just be part of the init for those modules:
```
  return {
    init: _.once(function () {
      WORKAREA.paypal.ready(function() {
        / * ... */
      });
    })
  }
```

Reply:
That would still pose the problem of the external `<script>` not being loaded in time, as the module `init()` functions will be run _after_ PayPal's `init()`.

Reply:
Oh I see the confusion - if I call `WORKAREA.paypal.ready` _after_ PayPal has been loaded, it won't run what I've passed. I think that's a fair expectation of how a `ready` works. jQuery and the affirm API are two examples.